pkg/cloud/aws/s3/client.go

```diff
@@ -0,0 +1,56 @@
+package s3
+
+import (
+	"github.com/applike/gosoline/pkg/cfg"
+	"github.com/applike/gosoline/pkg/cloud"
+	"github.com/applike/gosoline/pkg/mon"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/aws/aws-sdk-go/service/s3/s3iface"
+	"sync"
+)
+
+//go:generate mockery --name Client
+type Client interface {
+	s3iface.S3API
+}
+
+var c = struct {
+	sync.Mutex
+	instance map[string]*s3.S3
+}{}
+
+type Settings struct {
+	BasePath string
+	BucketName string
+	Client cloud.ClientSettings
+}
+
+func ProvideClient(config cfg.Config, logger mon.Logger, settings *Settings) *s3.S3 {
+	c.Lock()
+	defer c.Unlock()
+
+	if c.instance == nil {
+		c.instance = map[string]*s3.S3{}
+	}
+
+	endpoint := config.GetString("aws_s3_endpoint")
+	if c.instance[endpoint] != nil {
+		return c.instance[endpoint]
+	}
+
+	c.instance[endpoint] = NewClient(config, logger, settings)
+
+	return c.instance[endpoint]
+}
+
+func NewClient(config cfg.Config, logger mon.Logger, settings *Settings) *s3.S3 {
+	s3ForcePathStyle := config.GetBool("aws_s3_forcePathStyle", true)
```

Comment:
By default we should stick to the providers default, which would be false if i'm not mistaken.

```suggestion
	s3ForcePathStyle := config.GetBool("aws_s3_forcePathStyle", false)
```